skills/slashCommands.js

```diff
@@ -62,39 +62,42 @@ const getMessageFromRes = (res, learn) => {
     (a, b) => b - a
   );
 
-  const firstPlace = res.filter(row => row.count === uniqueSortedPoints[0]);
-  const secondPlace =
-    uniqueSortedPoints.length >= 2 &&
-    res.filter(row => row.count === uniqueSortedPoints[1]);
-  const thirdPlace =
-    uniqueSortedPoints.length >= 3 &&
-    res.filter(row => row.count === uniqueSortedPoints[2]);
-
-  let message = `*:tada: Congratulations :tada: to our best ${
-    learn ? 'learners' : 'teachers'
-  } this week*:
-
-  *${firstPlace.map(obj => obj.username).join(', ')}* ${
-  learn ? 'learned' : 'taught'
-} *${uniqueSortedPoints[0]}* ${
-  uniqueSortedPoints[0] === '1' ? 'thing' : 'things'
-}`;
-
-  if (secondPlace) {
-    message += `, *${secondPlace.map(obj => obj.username).join(', ')}* ${
-      learn ? 'learned' : 'taught'
-    } *${uniqueSortedPoints[1]}* ${
-      uniqueSortedPoints[1] === '1' ? 'thing' : 'things'
-    }`;
+  const role = learn ? 'learners' : 'teachers';
+  let message = `*:tada: Congratulations :tada: to our best ${role} this week*:\n\n`;
+
+  const scores = getScores(res, uniqueSortedPoints);
+  scores.forEach(score => {
+    message += getIndividualMessage(score, learn);
+  });
+
+  return message;
```

Comment:
> I'm typically against this kind out out-of-scope variable modification.

Does this refer to the `scores.forEach(...)` code? If so, why? It's not as compact, but seems to be the simpler, more "beginner" of doing it. Is it just that it could be better, or is there a reason why it's bad / dangerous?

Reply:
Two main reasons.
You had to define it as a `let` which means it can be accidentally edited later (by unfortunate name collisions or programmer error)
Also because it's easier for variables like that to "hang around" when they shouldn't and when they are suddenly processing an array of types of scores you can run into issues of the last type's data being left behind. 

Seen it all to many times. Neither of those are possible by not allowing the state to bleed out + using a `const`.

Reply:
Fair. Will consider this going forward :+1:

Also, for your second suggestion I get an eslint error:

`69:33  error  Arrow function should not return assignment   no-return-assign`

Figured you might be interested, I'll need to stick to your first suggestion because of it.

Reply:
That's ESLint's opinion and not necessarily mine. You could also work around it by:
```javascript

const message = scores.reduce((accum, score) => 
  `${accum}${getIndividualMessage(score, learn)}`, 
  `*:tada: Congratulations :tada: to our best ${role} this week*:\n\n`
);
```